yarn/src/main/scala/org/apache/spark/deploy/yarn/ClientArguments.scala

```diff
@@ -103,11 +104,15 @@ private[spark] class ClientArguments(args: Array[String], sparkConf: SparkConf)
           userClass = value
           args = tail
 
+        case ("--primaryResource") :: value :: tail =>
```

Comment:
Looked deeper and realized this isn't a spark-submit argument and that it's just used for the YARN stuff internally.

primaryResource is kind of a confusing name because a userJar is also a primary resource.  Maybe "main-py-module" instead?